### ISSUE TITLE
Add max retries to quorum to avoid live locks

### ISSUE THREAD

**Issue by d4l3k** [pytorch/torchft#151]
When we hit issues during training we can sometimes hit issues that are not recoverable and consistently fail but we retry. We should add an option to Manager `max_retries` which allows the user to say only retry `should_commit` up to N times before crashing and allowing the underlying scheduler to retry.

This should help avoid live lock behavior.

Relevant code:

* Manager.__init__ https://github.com/pytorch/torchft/blob/main/torchft/manager.py#L88
* Where we should check for should_commit success https://github.com/pytorch/torchft/blob/main/torchft/manager.py#L631
* manager mocked tests https://github.com/pytorch/torchft/blob/main/torchft/manager_test.py

Testing:

This should be fairly straightforward to add a manager test that uses the mocked manager and verifies that after N failed `should_commit` calls it throws an exception.
